Sources/SwiftKuery/ForeignKey.swift

```diff
@@ -0,0 +1,92 @@
+/**
+ Copyright IBM Corporation 2016, 2017, 2018
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import Foundation
+
+struct ForeignKey: Hashable {
+    var keyColumns: [Column]
+    var refColumns: [Column]
+    var keyNames: [String]
+    var refNames: [String]
+
+    public init?(keys: [Column], refs: [Column],_ tableName: String, _ errorString: inout String) {
+        if !ForeignKey.validKey(keys, refs, tableName, &errorString) {
+            return nil
+        }
+        keyColumns = keys
+        refColumns = refs
+        keyNames = keyColumns.map { "\($0._table._name).\($0.name)"}
```

Comment:
Issue with spaces

Reply:
I corrected the spaces at the end of the closure.

If you were referring to the need to pack its unnecessary here as these values will not be used in sql statements.